### PR TITLE
LibWeb: Implement linear easing according to latest spec

### PR DESCRIPTION
**Pull request by Gingeh**
Adds support for `linear` easing functions with stop lists. (e.g. `linear(0, 0.2 25% 75%, 1)`)

Also includes:
- A test case for the output values of easing functions.
- The easing function implementations have been split into their own methods.
- Easing function parsing is now case insensitive.
- `Vector::last_matching` has been made const (so I could use it)

The test has some commented-out cases because they're currently problematic:
- Output progress values outside the [0-1] range aren't handled by `StyleComputer::collect_animation_into`.
  I've added a temporary `clamp` so my tests don't hit undefined behaviour from an underflow.
- `Steps::Position::Start` isn't handled (should be equivalent to `Steps::Position::JumpStart`).
  This is a simple one line fix, should I include it in this PR or make another?

### CODE REVIEW

File: Userland/Libraries/LibWeb/CSS/StyleValues/EasingStyleValue.cpp [hunk 1]
```diff
@@ -57,30 +64,154 @@ bool EasingStyleValue::CubicBezier::operator==(Web::CSS::EasingStyleValue::Cubic
...
+    // 4. If any control point still lacks an input progress value,
+    // then for each contiguous run of such control points,
+    // set their input progress values so that they are evenly spaced
+    // between the preceding and following control points with input progress values.
+    Optional<size_t> run_start_idx;
+    for (size_t idx = 0; idx < stops.size(); idx++) {
+        auto stop = stops[idx];
+        if (stop.input.has_value() && run_start_idx.has_value()) {
+            // this stop is immediately after a run
+            // set inputs of [start, idx-1] stops to be evenly spaced between start-1 and idx
```
Comment (AtkinsSJ) [inline Userland/Libraries/LibWeb/CSS/StyleValues/EasingStyleValue.cpp:103]: To make it clear what's spec and what isn't, please prefix these with `Note:` or `NOTE:`. (I think we use both which is maybe not ideal - try whichever one the spec itself doesn't use. 😅)

File: Userland/Libraries/LibWeb/CSS/StyleValues/EasingStyleValue.cpp [hunk 2]
```diff
@@ -57,30 +64,154 @@ bool EasingStyleValue::CubicBezier::operator==(Web::CSS::EasingStyleValue::Cubic
...
+            auto start_input = stops[run_start_idx.value() - 1].input.value();
+            auto end_input = stops[idx].input.value();
+            auto run_stop_count = idx - run_start_idx.value() + 1;
+            auto delta = (end_input - start_input) / run_stop_count;
+            for (size_t run_idx = 0; run_idx < run_stop_count; run_idx++) {
+                stops[run_idx + run_start_idx.value() - 1].input = start_input + delta * run_idx;
+            }
+            run_start_idx = {};
+        } else if (!stop.input.has_value() && !run_start_idx.has_value()) {
+            // this stop is the start of a run
```
Comment (AtkinsSJ) [inline Userland/Libraries/LibWeb/CSS/StyleValues/EasingStyleValue.cpp:113]: Same point about comments

**Review by AtkinsSJ — CHANGES_REQUESTED** [review]
Just a nit-pick!

File: Userland/Libraries/LibWeb/CSS/StyleValues/EasingStyleValue.cpp (hunk 1 above)
Comment (Gingeh) [reply Userland/Libraries/LibWeb/CSS/StyleValues/EasingStyleValue.cpp]: Done :+1:

**Review by AtkinsSJ — APPROVED** [review]

**Pull request merged**